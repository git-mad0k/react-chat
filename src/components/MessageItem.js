import React from 'react';

import { withStyles } from '@material-ui/core/styles';
import Paper from '@material-ui/core/Paper';
import classnames from 'classnames';
import moment from 'moment';
import AvatarProfile from './AvatarProfile';

const styles = () => ({
  messageAuthor: {
    color: 'rgb(96, 125, 139)',
    fontSize: '1em',
    fontWeight: '400',
  },
  messageText: {
    fontSize: '1em',
    fontWeight: '400',
  },
  messageDate: {
    color: 'rgb(96, 125, 139)',
    fontSize: '1em',
    fontWeight: '400',
  },
  messageContainer: {
    marginLeft: '1rem',
    padding: '0.5rem',
    fontSize: '0.875rem',
    minWidth: '70px',
    maxWidth: '450px',
  },

  messageContainerForMe: {
    marginRight: '1rem',
    backgroundColor: '#4dc5ff3d',
  },
  messageWrapper: {
    display: 'flex',
    margin: '1rem 0.8rem',
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    alignItems: 'center',
  },

  messageWrapperForMe: {
    alignSelf: 'flex-end',
    flexDirection: 'row-reverse',
  },
  userStatusWrapper: {
    alignSelf: 'center',
    display: 'flex',
    margin: '1rem 0.8rem',
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    fontSize: '0.875rem',
    fontWeight: '400',
    textAlign: 'center',
    flexDirection: 'column',
  },
  userName: {
    fontSize: '1em',
    color: 'green',
  },
  userStatus: {
    fontSize: '1em',
    color: '#000',
  },
  userDateVisit: {
    fontSize: '0.875em',
    color: 'rgb(96, 125, 139)',
  },
});

const MessageItem = ({
  classes, message, key, activeUser,
}) => {
  /* eslint-disable  */
  const isMessageForMe = message.sender._id === activeUser._id;
  const colorForMe = message.sender._id === activeUser._id === 'me' ? 'red' : '';
  /* eslint-enable  */
  if (message.statusMessage) {
    return (
      <div className={classes.userStatusWrapper}>
        <div className={classes.userName}>
          {message.sender.username}
          {' '}
          <span className={classes.userStatus}>{message.content}</span>
        </div>
        <div className={classes.userDateVisit}>
          {' '}
          {moment(message.createdAt).fromNow()}
          {' '}
        </div>
      </div>
    );
  }

  const {
    messageWrapper,
    messageContainer,
    messageWrapperForMe,
    messageContainerForMe,
  } = classes;
  return (
    <div
      key={key}
      className={classnames(messageWrapper, isMessageForMe && messageWrapperForMe)}
    >
      <AvatarProfile name={message.sender.username} color={colorForMe} />
      <Paper
        className={classnames(messageContainer, isMessageForMe && messageContainerForMe)}
      >
        <div className={classes.messageAuthor}>{message.sender.username}</div>
        <div className={classes.messageText}>{message.content}</div>
        <div className={classes.messageDate}>{moment(message.date).fromNow()}</div>
      </Paper>
    </div>
  );
};

export default withStyles(styles)(MessageItem);
