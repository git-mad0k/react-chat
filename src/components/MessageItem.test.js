/* eslint-env jest */
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import MessageItem from './MessageItem';

const mockData = {
  message: {
    _id: '1',
    updatedAt: '2018-03-16T10:53:23.200Z',
    createdAt: '2018-03-16T10:53:23.200Z',
    content: 'test content 1',
    sender: {
      _id: '1',
      username: 'test2',
      lastName: 'test',
      firstName: 'test',
    },
  },

  activeUser: {
    _id: '1',
    firstName: 'test',
    lastName: 'test',
    username: 'test',
    isMember: false,
    isCreator: false,
    isChatMember: false,
  },
};

describe('MessageItem', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<MessageItem {...mockData} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('render corretly ', () => {
    const tree = renderer.create(<MessageItem {...mockData} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('redner no user message', () => {
    const tree = renderer
      .create(<MessageItem {...mockData} activeUser={{ ...mockData.activeUser, _id: '2' }} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('render user is joined ', () => {
    const tree = renderer
      .create(<MessageItem
        {...mockData}
        message={{ ...mockData.message, content: 'joined', statusMessage: true }}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
