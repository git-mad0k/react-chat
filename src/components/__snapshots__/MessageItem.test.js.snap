// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageItem redner no user message 1`] = `
<div
  className="MessageItem-messageWrapper-47"
>
  <div
    className="MuiAvatar-root-53 MuiAvatar-colorDefault-54"
    style={
      Object {
        "backgroundColor": "#2196f3",
      }
    }
  >
    T
  </div>
  <div
    className="MuiPaper-root-56 MuiPaper-elevation2-60 MuiPaper-rounded-57 MessageItem-messageContainer-45"
  >
    <div
      className="MessageItem-messageAuthor-42"
    >
      test2
    </div>
    <div
      className="MessageItem-messageText-43"
    >
      test content 1
    </div>
    <div
      className="MessageItem-messageDate-44"
    >
      a few seconds ago
    </div>
  </div>
</div>
`;

exports[`MessageItem render corretly  1`] = `
<div
  className="MessageItem-messageWrapper-47 MessageItem-messageWrapperForMe-48"
>
  <div
    className="MuiAvatar-root-53 MuiAvatar-colorDefault-54"
    style={
      Object {
        "backgroundColor": "#2196f3",
      }
    }
  >
    T
  </div>
  <div
    className="MuiPaper-root-56 MuiPaper-elevation2-60 MuiPaper-rounded-57 MessageItem-messageContainer-45 MessageItem-messageContainerForMe-46"
  >
    <div
      className="MessageItem-messageAuthor-42"
    >
      test2
    </div>
    <div
      className="MessageItem-messageText-43"
    >
      test content 1
    </div>
    <div
      className="MessageItem-messageDate-44"
    >
      a few seconds ago
    </div>
  </div>
</div>
`;

exports[`MessageItem render user is joined  1`] = `
<div
  className="MessageItem-userStatusWrapper-49"
>
  <div
    className="MessageItem-userName-50"
  >
    test2
     
    <span
      className="MessageItem-userStatus-51"
    >
      joined
    </span>
  </div>
  <div
    className="MessageItem-userDateVisit-52"
  >
     
    8 months ago
     
  </div>
</div>
`;
